geopandas/geoseries.py

```diff
@@ -534,7 +534,31 @@ def _constructor(self):
     def _constructor_expanddim(self):
         from geopandas import GeoDataFrame
 
-        return GeoDataFrame
+        def expanddim(data=None, index=None, crs=None, **kwargs):
+            # if geometry is somehow supplied, ignore it because we don't need it
+            kwargs.pop("geometry", None)
+            if isinstance(data, GeoSeries):
+                #  if GeoSeries has no name, we make the geometry name geometry
+                #  (instead of 0)
+                if data.name is None:
+                    data = {"geometry": data}
+                    geo_col_name = "geometry"
+                else:
+                    geo_col_name = data.name
+            else:  # Dict of {name : (Geo)Series pairs}
+                # if only one key, this is not ambiguous, if more than one
+                # i.e. from pd.concat([GeoSeries, GeoSeries]) the geom col comes
+                # from the left (same as for multiple gdfs in concat)
+                geo_col_name = list(data.keys())[0]
```

Comment:
Check here if this is `None`, and in that case use "geometry" as you did above?

Reply:
This makes sense, looks like it could be relevant for `NDFrame.align`? 

pd.concat is an odd case because `data = dict(zip(range(len(self.objs)), self.objs))` is how this dict gets defined, so the keys are an integer sequence and the column names get mapped on later. This relates to your comment later about why there's a case of a broken GeoDataFrame.

Reply:
I actually don't think I can do this for that very reason, in the case where `data` is a dictionary, any changes to the keys are going to get lost - and more than that, the insertion order of the dictionary is important -as it relates which columns get mapped to which names:
https://github.com/pandas-dev/pandas/blob/e39ea3024cebb4e7a7fd35972a44637de6c41650/pandas/core/reshape/concat.py#L550-L559

I don't see an elegant way around this. Perhaps the alternative approach is to use expanddim to construct a `pd.DataFrame` and then turn that into a GeoDataFrame, rather than trying to catch this case? (Potentially also helping with below?)